module_utils/argspec/base.py

```diff
@@ -0,0 +1,24 @@
+from ansible.module_utils.six import iteritems
+
+
+class ArgspecBase(object):
```

Comment:
If you are planning to provide direct attribute access to dict values, you should implement `__delattr__` as well.

Reply:
How will other module arguments be implemented?  `required_if`, `required_together`, etc

Reply:
@privateip Thanks for the review. So Base class code is actually not is use anymore, it's the old code. I have updated the Base class to be empty now. This is actually going to be achieved by the code generator.

Other module arguments such as `required_if` would be used the following way:
```
diff --git a/library/nxos_interfaces.py b/library/nxos_interfaces.py
index a636aa6..e902e50 100644
--- a/library/nxos_interfaces.py
+++ b/library/nxos_interfaces.py
@@ -50,6 +50,7 @@ def main():
     """ main entry point for module execution
     """
     module = AnsibleModule(argument_spec=Interface.argument_spec,
+                           required_if=Interface.required_if,
                            supports_check_mode=True)
 
     result = Interface(module).execute_module()
diff --git a/module_utils/argspec/nxos/interfaces/interfaces.py b/module_utils/argspec/nxos/interfaces/interfaces.py
index a97550e..62cefd6 100644
--- a/module_utils/argspec/nxos/interfaces/interfaces.py
+++ b/module_utils/argspec/nxos/interfaces/interfaces.py
@@ -19,3 +19,5 @@ class InterfaceArgs(ArgspecBase):
         'state': dict(default='merged', choices=['merged', 'replaced', 'overriden', 'deleted']),
         'config': dict(type='list', elements='dict', options=config_spec)
     }
+
+    required_if = [('state', 'merged', ('mode',))]
```